lms/djangoapps/courseware/courses.py

```diff
@@ -376,15 +376,15 @@ def get_courses(user, org=None, filter_=None):
     Returns a list of courses available, sorted by course.number and optionally
     filtered by org code (case-insensitive).
     """
+    include_hidden = filter_.pop('include_hidden', False) if filter_ else False
     courses = branding.get_visible_courses(org=org, filter_=filter_)
 
     permission_name = configuration_helpers.get_value(
         'COURSE_CATALOG_VISIBILITY_PERMISSION',
         settings.COURSE_CATALOG_VISIBILITY_PERMISSION
     )
 
-    courses = [c for c in courses if has_access(user, permission_name, c)]
-
+    courses = [c for c in courses if include_hidden or has_access(user, permission_name, c)]
```

Comment:
Agreed.  Want to confirm that you can't access courses you wouldn't already be able to access just by passing the `include_hidden` flag